### PR TITLE
Add molecular structure types to cake example

### PR DESCRIPTION
**Pull request by kroenlein**
Molecular structure types were added to the cake example.  In particular, for the two items that already had expected molecular structures, a SMILES was added to sugar and an InChI was added to salt, both using the same Molecular Structure bounds object.

This is expected to be a proper release.

https://citrine.atlassian.net/browse/PLA-3934

### CODE REVIEW

File: gemd/demo/cake.py [hunk 1]
```diff
@@ -113,6 +115,11 @@ def make_cake_templates():
         description="The chemical formula of a material",
         bounds=CompositionBounds(components=EmpiricalFormula.all_elements())
     )
+    tmpl["Molecular Structure"] = PropertyTemplate(
+        name="Molecular Structure",
+        description="The molecular structure of the material",
+        bounds=MolecularStructureBounds()
+    )
```
Comment (bfolie) [inline gemd/demo/cake.py:122]: How is this template used?

Comment (kroenlein) [reply gemd/demo/cake.py:122]: It appears (or should be appearing) as a PropertyAndCondition on the MaterialSpec for salt and for sugar.  The former is represented as an InChI, the latter as a SMILES.

Comment (bfolie) [reply gemd/demo/cake.py:122]: How are the `PropertyAndCondition` objects linked back to the template? I see, for example,
`Property(name='SMILES', value=Smiles("C(C1C(C(C(C(O1)OC2(C(C(C(O2)CO)O)O)CO)O)O)O)O")))`, but shouldn'T that property be linked to the `PropertyTemplate`?

Comment (kroenlein) [reply gemd/demo/cake.py:122]: Yes.  Yes it should.

**Pull request merged**